src/test/pagination.ts

```diff
@@ -0,0 +1,208 @@
+    expect(usersResponse.data.users.users[3].name).to.be.equal('De Code');
+    expect(usersResponse.data.users.users[3].email).to.be.equal('user1@taq.com');
+    expect(usersResponse.data.users.users[4].name).to.be.equal('Ze Dev');
+    expect(usersResponse.data.users.users[4].email).to.be.equal('user2@taq.com');
+  });
+
+  it('should return shifted list of 3 users: with previous, with next', async () => {
+    const variables = { step: 1 };
+    const usersResponse = await usersResponseData(token, variables);
+
+    expect(usersResponse.data.users.users).to.be.sortedBy('name');
+    expect(usersResponse.data.users.users.length).to.be.equal(3);
+    expect(usersResponse.data.users.totalUsers).to.be.equal(dbSize);
+    expect(usersResponse.data.users.previous).to.be.equal(true);
+    expect(usersResponse.data.users.next).to.be.equal(true);
+
+    expect(usersResponse.data.users.users[0].name).to.be.equal('B Usuario');
+    expect(usersResponse.data.users.users[0].email).to.be.equal('user0@taq.com');
+    expect(usersResponse.data.users.users[1].name).to.be.equal('Da Taq');
+    expect(usersResponse.data.users.users[1].email).to.be.equal('user3@taq.com');
+    expect(usersResponse.data.users.users[2].name).to.be.equal('De Code');
+    expect(usersResponse.data.users.users[2].email).to.be.equal('user1@taq.com');
+  });
+
+  it('should return list of last 3 users: with previous, no next', async () => {
+    const variables = { step: 2 };
+    const usersResponse = await usersResponseData(token, variables);
+
+    expect(usersResponse.data.users.users).to.be.sortedBy('name');
+    expect(usersResponse.data.users.users.length).to.be.equal(3);
+    expect(usersResponse.data.users.totalUsers).to.be.equal(dbSize);
+    expect(usersResponse.data.users.previous).to.be.equal(true);
+    expect(usersResponse.data.users.next).to.be.equal(false);
+
+    expect(usersResponse.data.users.users[0].name).to.be.equal('Da Taq');
+    expect(usersResponse.data.users.users[0].email).to.be.equal('user3@taq.com');
+    expect(usersResponse.data.users.users[1].name).to.be.equal('De Code');
+    expect(usersResponse.data.users.users[1].email).to.be.equal('user1@taq.com');
+    expect(usersResponse.data.users.users[2].name).to.be.equal('Ze Dev');
+    expect(usersResponse.data.users.users[2].email).to.be.equal('user2@taq.com');
+  });
+
+  it('should return shifted list of users bigger than db: with previous, no next', async () => {
+    const variables = { maxUsers: 10, step: 3 };
+    const usersResponse = await usersResponseData(token, variables);
+
+    expect(usersResponse.data.users.users).to.be.sortedBy('name');
+    expect(usersResponse.data.users.users.length).to.be.equal(2);
+    expect(usersResponse.data.users.totalUsers).to.be.equal(dbSize);
+    expect(usersResponse.data.users.previous).to.be.equal(true);
+    expect(usersResponse.data.users.next).to.be.equal(false);
+
+    expect(usersResponse.data.users.users[0].name).to.be.equal('De Code');
```

Comment:
não daria pra usar o 
`expect(usersResponse.data.users.user[0]).to.be.deep.eq({ ... })` para comparar tudo? dado que vc ja tem o que ta no banco